res/generate_update_metadata.py

```
@@ -0,0 +1,307 @@
+        fail(f"release id must not require URL decoding: {release_id}")
+def display_version_from_release_id(release_id):
+    match = re.fullmatch(r"v?(\d+)\.(\d+)\.(\d+)", release_id)
+    if not match:
+        return None
+    version_segments = [int(part) for part in match.groups()]
+    if any(segment > 65535 for segment in version_segments):
+        fail(f"release id has out-of-range display version: {release_id}")
+    if all(segment == 0 for segment in version_segments):
+        fail(f"release id has invalid display version: {release_id}")
+    return ".".join(match.groups())
+def validate_release_version(release_id, version, print_skip=False):
```

Ruff (0.16.1)

[warning] Boolean default positional argument in function definition

(FBT002)